daemon/top_unix.go

```diff
@@ -135,12 +136,20 @@ func psPidsArg(pids []uint32) string {
 // "-ef" if no args are given.  An error is returned if the container
 // is not found, or is not running, or if there are any problems
 // running ps, or parsing the output.
-func (daemon *Daemon) ContainerTop(name string, psArgs string) (*container.ContainerTopOKBody, error) {
-	if psArgs == "" {
-		psArgs = "-ef"
+// to deal with `docker top redisstor -C "sto redis-server"`,
+// we change the second arg to json format
+func (daemon *Daemon) ContainerTop(name string, psArgsJSON string) (*container.ContainerTopOKBody, error) {
+	var args []string
+	if err := json.Unmarshal([]byte(psArgsJSON), &args); err != nil {
+		// It's just to be compatible with docker/cli before use the latest interface
+		// TODO: when docker/cli use the latest interface, here can return the err
```

Comment:
We can't remove the fallback, because CLI and daemon don't have to be the same version (i.e., a client can connect to any remote daemon, which may have any version)

Reply:
Yes, I'll remove this TODO. Thanks.